Yosemite/Yosemite/Actions/InboxNotesAction.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import Networking
+
+// MARK: - InboxNotesAction: Defines Inbox Notes operations.
+//
+public enum InboxNotesAction: Action {
+
+    /// Retrieves all of the `InboxNote`s from the API.
+    ///
+    case loadAllInboxNotes(siteID: Int64,
+                           pageNumber: Int,
+                           pageSize: Int,
+                           orderBy: InboxNotesRemote.OrderBy,
+                           type: [InboxNotesRemote.NoteType]?,
+                           status: [InboxNotesRemote.Status]?,
+                           completion: (Result<[InboxNote], Error>) -> ())
+
+    /// Dismiss one `InboxNote`.
+    /// This internally marks a notification’s is_deleted field to true and such notifications do not show in the results anymore.
```

Comment:
just a note for the future: if we decide not to persist the notes, we might want to manually hide the note on this action. Otherwise, we'd have to sync the notes again